legacy/include/distconv/tensor/halo_cuda.hpp

```diff
@@ -655,9 +655,7 @@ void TraverseHalo(Tensor &tensor, int dim,
                   bool inner, OpType op,
                   cudaStream_t s) {
   // ConstDataType is const DataType if the operation only reads the tensor.
-  using ConstDataType = typename std::conditional<
-    OpType::modifies_tensor, typename Tensor::data_type,
-    typename std::add_const<typename Tensor::data_type>::type>::type;
+  using ConstDataType = typename OpType::const_data_type;
```

Comment:
Do we still uses modifies_tensor now?

Reply:
It may be misleading since `OpType::const_data_type` is not always `const`. Perhaps moving it to `Tensor` and making:
```suggestion
  using ConstDataType = typename std::conditional<
    OpType::modifies_tensor,
    typename Tensor::data_type,
    typename Tensor::const_data_type
  >
```
This is assuming `std::conditional` is being cooperative in NVCC.

Reply:
Neither is `ConstDataType`...